recipes/boost/all/conanfile.py

```diff
@@ -1161,7 +1161,8 @@ def _ranlib(self):
 
     @property
     def _cxx(self):
-        cxx = self.buildenv.vars(self).get("CXX")
+        compilers_by_conf = self.conf.get("tools.build:compiler_executables", default={}, check_type=dict)
+        cxx = compilers_by_conf.get("cpp") or VirtualBuildEnv(self).vars().get("CXX")
```

Comment:
I am not sure the 2.0 stuff should move forward with env vars or us the compilers from conf 🤔 but I think we can wait to see how that recommendation falls out

This matches with the explanation I got about how this feature works